frontend/src/js/components/settings/organization/Billing.test.tsx

```diff
@@ -0,0 +1,116 @@
+import { render } from '../../../../../tests/setupTests';
+import { Billing } from './Billing';
+
+const preloadedState = {
+  organization: {
+    ...defaultState.organization,
+    organization: {
+      ...defaultState.organization.organization,
+      billing_profile: {
+        email: 'ok@ok.ok',
+        name: 'Test account',
+        address: {
+          country: 'NO',
+          city: 'test City',
+          line1: 'Test address',
+          postal_code: '1234',
+          state: 'test State'
+        },
+        shipping: {
+          name: 'Test account',
+          address: {
+            country: 'NO',
+            city: 'test City',
+            line1: 'Test address',
+            postal_code: '1234',
+            state: 'test State'
+          }
+        }
+      }
+    },
+    card: {
+      last4: '1234',
+      expiration: { month: 8, year: 1230 },
+      brand: 'Visa'
+    }
+  }
+};
+
+const editProfileActionParams = {
+  billingProfile: {
+    address: {
+      city: 'Oslo',
+      country: 'PL',
+      line1: 'Blindernveien',
+      postal_code: '5678',
+      state: 'Oslo'
+    },
+    email: 'ok@ok.ok',
+    name: 'Test account'
+  }
+};
+
+describe('Billing Component', () => {
+  it('renders correctly', async () => {
+    const { baseElement } = render(<Billing />, {});
+    const view = baseElement;
+    expect(view).toMatchSnapshot();
+    expect(view).toEqual(expect.not.stringMatching(undefineds));
+  });
+
+  it('supports modifying billing profile', async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    const ui = <Billing />;
+    const editProfileAction = jest.spyOn(OrganizationActions, 'editBillingProfile');
+    render(ui, { preloadedState });
+    expect(screen.getByText(/1234, test city/i)).toBeVisible();
+    await act(async () => await user.click(screen.getByRole('button', { name: /edit/i })));
+    const input = screen.getByLabelText<HTMLInputElement>('Country');
+
+    await act(async () => {
+      const addressInput = screen.getByRole('textbox', { name: /address line 1/i });
+      const stateInput = screen.getByRole('textbox', { name: /state/i });
+      const cityInput = screen.getByRole('textbox', { name: /city/i });
+      const zipInput = screen.getByRole('textbox', { name: /zip or postal code/i });
+      await user.clear(addressInput);
+      await user.clear(stateInput);
+      await user.clear(cityInput);
+      await user.clear(zipInput);
+
+      await user.type(addressInput, 'Blindernveien');
+      await user.type(stateInput, 'Oslo');
+      await user.type(cityInput, 'Oslo');
+      await user.type(zipInput, '5678');
+
+      const countryAutoComplete = screen.getByRole('combobox', { name: /country/i });
+
+      await user.type(countryAutoComplete, 'Polan');
+      await user.keyboard('[ArrowUp]');
+      await user.keyboard('[Enter]');
+      await user.click(screen.getByRole('button', { name: /save/i }));
+    });
```

Comment:
this should not be required and is recommended against, as the user actions get wrapped by act internally by RTL already